### PR TITLE
Make --to-ns and --ns-hosts work well together

### PR DESCRIPTION
**Pull request by cbrianhill**
I am loving using sshuttle as my poor person's VPN.  I use the --to-ns and the --ns-hosts options along with --dns and domain-specific resolver files in /etc/resolver to route only private DNS zones over the connection, and discovered that it didn't seem to work.

This fixed the problem for me.

### CODE REVIEW

File: sshuttle/server.py [hunk 1]
```diff
@@ -192,6 +192,8 @@ def try_send(self):
             port = 53
             family, sockaddr = self._addrinfo(peer, port)
         else:
+            _, peer = self.to_nameserver
+            port = 53
```
Comment (brianmay) [inline sshuttle/server.py]: Unfortunately, we don't actually know that the port number is 53. Also peer isn't a simple text string, it is a list of tuples returned by `getaddrinfo`.

A better option, maybe we need to preserve the values of `peer` and `port` in `self` after line 174.

Does this make sense?

**Comment by brianmay** [comment]
Thanks for the pull request. Does look like I stuffed up this use case.

**Comment by brianmay** [comment]
Thanks.

**Comment by cbrianhill** [comment]
Thanks to you for maintaining this great tool!  Glad to tidy up this one little use case :)

**Pull request merged**